iib/workers/dogpile_cache.py

```diff
@@ -0,0 +1,62 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import functools
+
+from dogpile.cache import make_region
+
+from iib.workers.config import get_worker_config
+
+
+def skopeo_inspect_should_use_cache(*args, **kwargs):
+    """Return true in case this requests can be taken from or stored in cache."""
+    return any(arg.find('@sha256:') != -1 for arg in args)
+
+
+def dogpile_cache(dogpile_region, should_use_cache_fn):
+    """
+    Dogpile cache decorator.
+
+    :params dogpile_region: Dogpile CacheRegion object
+    :params should_use_cache_fn: function which determines if cache should be used
+    """
+
+    def cache_decorator(func):
+        @functools.wraps(func)
+        def inner(*args, **kwargs):
+            should_cache = should_use_cache_fn(*args, **kwargs)
+            cache_key = generate_cache_key(func.__name__, *args, **kwargs)
+
+            if should_cache:
+                # get data from cache
+                output_cache = dogpile_region.get(cache_key)
+                if output_cache:
+                    return output_cache
+
+            output = func(*args, **kwargs)
+
+            if should_cache:
+                dogpile_region.set(cache_key, output)
+
+            return output
+
+        return inner
+
+    return cache_decorator
+
+
+def generate_cache_key(fn, *args, **kwargs):
+    """Generate key that is used in dogpile cache."""
+    arguments = "|".join(
+        [str(arg) for arg in args] + [f"{kwarg}={kwargs[kwarg]}" for kwarg in kwargs]
```

Comment:
I apologize for knit-picking but can we please use single quotes here? :)

Reply:
oh, I am just wondering why we have single quotes as standard... 
it is just not natural for me :D

Reply:
updated, thanks :)